### PR TITLE
feat(LinkingComponent): Adds link to new rules page, and a button to …

### PR DESCRIPTION
**Pull request by adonispuente**
This PR address adding a 'reset rules' button [RHICOMPL-1699](https://issues.redhat.com/browse/RHICOMPL-1699)
and adding a link to the newly created rules page as seen in these mocks https://marvelapp.com/prototype/93199gg/screen/88157324 

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: src/PresentationalComponents/TabbedRules/ProfileTabContent.js [hunk 1]
```diff
@@ -95,6 +99,7 @@ const ProfileTabContent = ({
   selectedRuleRefIds,
   rulesTableProps,
   newOsMinorVersion,
+  revealLinks,
```
Comment (bastilian) [inline src/PresentationalComponents/TabbedRules/ProfileTabContent.js:102]: Let's treat the links separately. That way we could in the future enable the reset feature in the Edit Policy, where we might not want to show the default ruleset link.

**Review by bastilian — APPROVED** [review]
Awesome work! This works like a charm! Thank you @adonispuente!

File: src/PresentationalComponents/TabbedRules/ProfileTabContent.js [hunk 2]
```diff
@@ -107,6 +112,14 @@ const ProfileTabContent = ({
...
   });
   const rules = handleSelect ? benchmark?.benchmark?.rules : profile?.rules;
+  const [originalRules, setOriginalRules] = useState([]);
+  useEffect(() => {
+    !loading ? setOriginalRules(selectedRuleRefIds) : {};
+  }, [loading]);
+
+  const resetRules = () => {
+    handleSelect && handleSelect(profile, newOsMinorVersion, originalRules);
+  };
```
Comment (bastilian) [inline src/PresentationalComponents/TabbedRules/ProfileTabContent.js:122]: A small optimisation we could make here is try to put the ResetLink in it's own component that has the effect and handler etc. and only render that component here when it is enabled.

File: src/PresentationalComponents/ResetRules/ResetRules.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  handleSelect,
+  updateRules,
+  profile,
+  newOsMinorVersion,
+  originalRules,
+  loading,
+  selectedRuleRefIds,
+}) => {
+  useEffect(() => {
+    !loading ? updateRules(selectedRuleRefIds) : {};
```
Comment (bastilian) [inline src/PresentationalComponents/ResetRules/ResetRules.js:15]: This should probably be something like 

```js
if(!loading) {
  updateRules(selectedRuleRefIds);
}
```

That's what LGTM is complaining about.

File: src/PresentationalComponents/TabbedRules/ProfileTabContent.js [hunk 4]
```diff
@@ -118,7 +126,32 @@ const ProfileTabContent = ({
             </span>
             <ProfileSystemCount count={systemCount} />
           </Text>
-          <SSGVersionText {...{ profile, newOsMinorVersion }} />
+          <Flex>
+            <FlexItem>
+              <SSGVersionText {...{ profile, newOsMinorVersion }} />
+            </FlexItem>
+            {resetLink && rulesPageLink && (
```
Comment (bastilian) [inline src/PresentationalComponents/TabbedRules/ProfileTabContent.js:133]: This should maybe be split up and used with each link individually.

Comment (adonispuente) [reply src/PresentationalComponents/TabbedRules/ProfileTabContent.js:133]: We'd only want the flex div to be there if one of them is valid, so we'd still need that check. 
I thought it looked a little messy but we can do something like this 

```
{(resetLink || rulesPageLink) && (
              <FlexItem align={{ default: 'alignRight' }}>
                {rulesPageLink && (
                  <Link
                    to={`/scappolicies/${profile?.id}/default_ruleset`}
                    target="_blank"
                    className="pf-u-mr-xl"
                  >
                    View policy rules
                    <ExternalLinkAltIcon className="pf-u-ml-sm" />
                  </Link>
                )}
                {resetLink && (
                  <ResetRules
                    handleSelect={handleSelect}
                    updateRules={setOriginalRules}
                    profile={profile}
                    newOsMinorVersion={newOsMinorVersion}
                    originalRules={originalRules}
                    loading={loading}
                    selectedRuleRefIds={selectedRuleRefIds}
                  />
                )}
              </FlexItem>
            )}
```

File: src/PresentationalComponents/ResetRules/ResetRules.js (hunk 3 above)
Comment (adonispuente) [reply src/PresentationalComponents/ResetRules/ResetRules.js:15]: ahhh nice catch

File: src/PresentationalComponents/TabbedRules/ProfileTabContent.js (hunk 4 above)
Comment (bastilian) [reply src/PresentationalComponents/TabbedRules/ProfileTabContent.js:133]: > We'd only want the flex div to be there if one of them is valid

Why?

Comment (adonispuente) [reply src/PresentationalComponents/TabbedRules/ProfileTabContent.js:133]: Then there could be scenarios where we just have an empty div aligned to the right

Comment (bastilian) [reply src/PresentationalComponents/TabbedRules/ProfileTabContent.js:133]: I think for the sake of readability we can tolerate the empty div. :)

Comment (adonispuente) [reply src/PresentationalComponents/TabbedRules/ProfileTabContent.js:133]: gotcha, ill send it up now

**Pull request merged**